lib/lightning/work_orders.ex

```diff
@@ -154,7 +154,7 @@ defmodule Lightning.WorkOrders do
     {:error, changeset}
   end
 
-  defp get_or_create_snapshot(multi, workflow \\ nil, name \\ :snapshot) do
+  defp get_or_create_snapshot(multi, workflow \\ nil, name \\ :snapshot, actor) do
```

Comment:
@rorymckinley would it be better/easier if we had two functions (or just a getter?), the reason this function is the way it is, is because when snapshots were first added - we gradually migrated everything using this function - but now every workflow has a snapshot. 

Since you've gone and done all the hard work of tracking down all these calls, perhaps we can make the assumption that snapshots exist and avoid passing down `actor` in all the "read" paths, like retrying, `create_for` with triggers etc?